ConnorAndersonLarson/Intention-Timer
# Feature/this isnt even my final form

## Is this a fix or a feature?

This is a fix and a minor feature.

## What is the change?

Major refactoring across all files, aiming at improving readability and minimizing code where able. Also adds animation for the color circle to countdown with the timer. On completion the ring flashes once and animates back to full color.

## What does it fix?

Slims down various functions where able, cleans up code, makes HTML more screen-reader friendly.

## Where should the reviewer start?

Reviewers should first clone and test usability using open index.html in their web browser. Test for functionality across all features (inputs, error messages, buttons work/don't work as expected, timer counts down properly, items are saved and restored on log activity button press/ refresh)

After this Reviewers should open and inspect code for errors/issues/places to slim down as desired. Files to look at are 
activity.js
index.html
main.js 
styles.css

## How should this be tested?

Testing can be done in web browser and running through the site functions. Basic usage includes:
Only one category button at once should light up on presses, unselecting should yield error
Leaving any input field empty should immediately yield errors
Putting non numbers into the minutes/seconds fields yield errors
Putting more than 59 seconds in the seconds field calculates minutes and adds to minutes fields/takes away from seconds field
Pressing Start Activity button before doing anything will yield errors on all empty/unselected categories
Start button in timer can only be pressed once
Start disappears on press
After timer runs down a complete message appears, cannot be pressed more than once.
Log activity button appears after timer runs.
Pressing log activity saves activity
Previous activities are saved on right hand side
Pressing Start new activity button brings up home page, can be repeated
Colors are changed on new activity (if first activity is purple and second activity is orange, all styling is orange instead of purple)
Reloading page saves activities on right hand side
